### PR TITLE
GFORMS-2066: Add ability to have 2 dataRetrieve calls on 1 page and the 2nd be conditional on the first

### PR DESCRIPTION
**Pull request by lxol**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/gform/processor/FormProcessor.scala [hunk 1]
```diff
@@ -269,53 +270,83 @@ class FormProcessor(
...
+            val maybeRequestParams = DataRetrieve.requestParamsFromCache(cache.form, n.id)
+            implicit val ni: NinoInsightsConnector[Future] = ninoInsightsConnector
+            DataRetrieveService[NinoInsights, Future]
+              .retrieve(n, visibilityOptics, maybeRequestParams)
+          case a: BankAccountInsights =>
+            val maybeRequestParams = DataRetrieve.requestParamsFromCache(cache.form, a.id)
+            implicit val ai: BankAccountInsightsConnector[Future] = bankAccountInsightConnector
+            val iff = a.`if`
+              .map(visibilityOptics.evalIncludeIfExpr(_, None))
+              .getOrElse(true)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/gform/processor/FormProcessor.scala:313]: This `if` should be generic for any data retrieve.

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/sharedmodel/DataRetrieve.scala [hunk 2]
```diff
@@ -249,7 +258,8 @@ object DataRetrieve {
   final case class BankAccountInsights(
     override val id: DataRetrieveId,
     sortCode: Expr,
-    accountNumber: Expr
+    accountNumber: Expr,
+    override val `if`: Option[IncludeIf] = None
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/DataRetrieve.scala:262]: Please remove default value from `if` (here and elsewhere)

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/gform/processor/FormProcessor.scala [hunk 3]
```diff
@@ -269,53 +270,81 @@ class FormProcessor(
...
-        } else Option.empty.pure[Future]
+          val initialVisibilityOptics = processData.formModelOptics.formModelVisibilityOptics
+          pageModel
+            .fold { singleton =>
+              singleton.page
+                .dataRetrieves()
+                .foldLeft(Future.successful(List.empty[DataRetrieveResult] -> initialVisibilityOptics)) {
+                  case (acc, r) =>
+                    acc.flatMap {
+                      case (results, optics) if r.`if`.map(optics.evalIncludeIfExpr(_, None)).getOrElse(true) =>
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/gform/processor/FormProcessor.scala:331]: Does this handle multiple transitively dependent DataRetrievals? I'm testing chain of 3 data retrievals and it evaluates them, even though theirs `if` should evaluate to `false`.

Comment (lxol) [reply app/uk/gov/hmrc/gform/gform/processor/FormProcessor.scala:331]: It should handle, will have a look

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**